server/server.go

```diff
@@ -84,8 +88,9 @@ func (s *Server) RunUntilShutdown() error {
 	}
 
 	httpServer := &http.Server{
-		Addr:    s.cfg.HTTPListener,
-		Handler: cors.New(corsOpts).Handler(s.newHttpHandler()),
+		Addr:              s.cfg.HTTPListener,
+		Handler:           cors.New(corsOpts).Handler(s.newHttpHandler()),
+		ReadHeaderTimeout: 30 * time.Second,
```

Comment:
what's the gosec error if we don't add this?

Reply:
> G112 (CWE-400): Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server

So a timeout must be added to shut this alarm

Reply:
ah this is an interesting attack:

"
A Slowloris attack is a type of Denial of Service (DoS) attack that targets web servers. The attacker sends HTTP requests very slowly, one partial header at a time, keeping connections open for as long as possible. This can exhaust the server's connection pool, making it unavailable for legitimate users."